### PR TITLE
Consume appcenter-file-upload-client-node as a package

### PR DESCRIPTION
**Pull request by russelarms**
The [original PR's]( https://github.com/microsoft/appcenter-cli/pull/978) description:

> It was decided to treat mc-fus as a separate library and publish it to npm without creating a separate repo.
> This PR makes it a proper library and moves to the root of the repo (for visibility).
> Also did some minor renaming to differentiate it with the original mc-fus library for web.
> This PR also adds an additional step to CI that builds the library.

The changes to the original PR:
- [x] fixed vulnerabilities in the ac-fus-lib dev dependencies
- [x] ac-fus-lib connected as an npm dependency in the main package.json, it should fix the [linux issue](https://github.com/microsoft/appcenter-cli/issues/1007)
- [x] ac-fus-lib types are imported from the npm dependency

Tested:
- [x] manually
- [x] with `npm test` and `npm test:ci`
- [x] ci tests

Related work items:
[AB#82467](https://msmobilecenter.visualstudio.com/Mobile-Center/_boards/board/t/IvanM-Team/Backlog%20Items/?workitem=82467)
[AB#82466](https://msmobilecenter.visualstudio.com/Mobile-Center/_boards/board/t/IvanM-Team/Backlog%20Items/?workitem=82466)

### CODE REVIEW

File: src/commands/distribute/lib/mc-fus-uploader/readme.md [hunk 1]
```diff
@@ -1,8 +0,0 @@
-This directory contains a NodeJS port of the library used to upload releases to App Center.
-
-The library leverages App Center File Upload Service and allows uploading many chunks (slices of a release binary) in parallel. It takes the binary file, slices it in 4mb chunks, and uploads them simultaneously.
```
Comment (MatkovIvan) [inline src/commands/distribute/lib/mc-fus-uploader/readme.md:3]: Should we keep some of this desc in the new readme?

File: package.json [hunk 2]
```diff
@@ -44,7 +46,10 @@
   "files": [
     "package-lock.json",
     "bin",
-    "dist"
+    "dist",
+    "appcenter-file-upload-client-node/out",
+    "appcenter-file-upload-client-node/package.json",
+    "appcenter-file-upload-client-node/package-lock.json"
```
Comment (MatkovIvan) [inline package.json:52]: This is wrong. These files must not be directly included in CLI npm package.

**Comment by jp-andre** [comment]
Thanks for working on this. There is still a build error with integration tests (the ones in powershell).

**Comment by annakocheshkova** [comment]
@jp-andre it has not appeared on the 2nd run. Probably a flaky one, since we're talking to the real AppCenter API.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -40,16 +40,36 @@ jobs:
...
+    workingDirectory: appcenter-file-upload-client-node
+
+  - task: PublishTestResults@2
+    inputs:
+      testResultsFormat: 'JUnit'
+      testResultsFiles: 'test-results.xml'
+      searchFolder: '$(Build.SourcesDirectory)/appcenter-file-upload-client-node'
+      failTaskOnFailedTests: true
+      testRunTitle: 'appcenter-file-upload-client-node unit tests'
+    displayName: 'Verify appcenter-file-upload-client-node unit test result'
```
Comment (jp-andre) [inline azure-pipelines.yml:72]: Could we have an "install test" here? Where we validate that `npm install -g` works on the package? Ideally we should be verifying the scenario that didn't work (linux with sudo and no nvm), but I'm not sure this is easily feasible within this pipeline.

File: package.json [hunk 4]
```diff
@@ -17,7 +17,8 @@
   },
   "preferGlobal": true,
   "scripts": {
-    "build": "npm run lint && gulp build:raw",
+    "build-ac-fus": "cd appcenter-file-upload-client-node && npm run build",
+    "build": "npm run build-ac-fus && npm run lint && gulp build:raw",
```
[the remark this answers is not in the document]
Comment (annakocheshkova) [reply package.json]: I don't think we need it. For CI, we have [this step](https://github.com/microsoft/appcenter-cli/pull/1021/files#diff-fec826feae04e51c0d94076385408bdcR55), similar to building the cli itself.

File: package.json [hunk 5]
```diff
@@ -103,6 +104,7 @@
   },
   "dependencies": {
     "abort-controller": "^3.0.0",
+    "appcenter-file-upload-client-node": "1.0.0",
```
[the remark this answers is not in the document]
Comment (annakocheshkova) [reply package.json:106]: I don't think they are somehow related.

File: azure-pipelines.yml (hunk 3 above)
Comment (annakocheshkova) [reply azure-pipelines.yml:72]: We have npm install [here](https://github.com/microsoft/appcenter-cli/pull/1021/files#diff-fec826feae04e51c0d94076385408bdcR55), and if it doesn't work, then testing step will fail. I don't see similar verification steps for the `npm i` for CLI so not sure why we need it?

Comment (jp-andre) [reply azure-pipelines.yml:72]: Sorry, I meant installing the appcenter-cli package itself onto the system. i.e. `npm pack && npm install -g appcenter-cli*.tgz`.
What was failing in the previous attempt was installing this package on a user's computer.

File: package.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (jp-andre) [reply package.json:106]: Okay. Too bad we did not coordinate those efforts better, but it's OK :)

File: appcenter-file-upload-client-node/package.json [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  "description": "File Uploader Client for Node",
+  "homepage": "https://github.com/microsoft/appcenter-cli",
+  "author": "Microsoft",
+  "license": "MIT",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/microsoft/appcenter-cli"
+  },
+  "scripts": {
+    "build": "npm run lint && npm run build:tsc",
```
Comment (jp-andre) [inline appcenter-file-upload-client-node/package.json:13]: I didn't mean to remove build-ac-fus entirely, only to remove it from build. But that's ok. :)

**Review by anywherepilot — APPROVED** [review]
This overall looks great to me. I would have put the .gitignore stuff in a separate file in the sub folder but the effect is the same. JP and I were thinking of adding a package installation test just as a followup to the incident that caused the revert of the previous version of this change. It's not related to the new package, but it would be a nice hardening followup to what was caused by a previous release of this change.

Dependabot is still enabled on the sub folder, and we'll handle those updates going forward. We'll also release new versions of the package as part of that process.

**Comment by jp-andre** [comment]
> This overall looks great to me. I would have put the .gitignore stuff in a separate file in the sub folder but the effect is the same. JP and I were thinking of adding a package installation test just as a followup to the incident that caused the revert of the previous version of this change. It's not related to the new package, but it would be a nice hardening followup to what was caused by a previous release of this change.

It's included in this PR as I requested it, though this is only a simple test here, that wouldn't have caught the linux-only sudo issue.

**Review by jp-andre — APPROVED** [review]

**Review by MatkovIvan — APPROVED** [review]

**Pull request merged**